### PR TITLE
AB Email Notifications - CSS High Priority Fix

### PR DESCRIPTION
**Pull request by HasnatL-Adapptlimited**
# Links 
[Base PR](https://github.com/ReliefApplications/ems-frontend/pull/2460)

In Reference to:
https://github.com/ReliefApplications/ems-frontend/pull/2460#discussion_r1575911242

# Tickets

- [ ] High Priority CSS Fix, removed custom css for some classes and replaced buttons with ui-button
- [ ] Button No Longer Shows Tick when clicking apply filter

Before:
image

### CODE REVIEW

**Comment by HasnatL-Adapptlimited** [comment]
Hi @AntoineRelief,
CC: @coian-AdapptLimited 

Apologies for the late push this PR has been opened for the High Priority code fix 
https://github.com/ReliefApplications/ems-frontend/pull/2460#discussion_r1575911242

File: libs/shared/src/lib/components/email/components/email-template/email-template.component.html [hunk 1]
```diff
@@ -411,87 +411,94 @@
...
-  </div>
-  <div uiChipList>
-    <ui-chip
-      variant="primary"
+    <div class="loadingSign" [ngClass]="{active: loading}">
+        <ui-spinner *ngIf="loading" style="position: relative;"></ui-spinner>
+      </div>
+    <div uiChipList>
+      <ui-chip
+      variant="secondary"
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/email/components/email-template/email-template.component.html:501]: @HasnatL-Adapptlimited 

it looks like the last version of the main branch wasn't used for this PR, as the code cannot build due to this line

No need to edit it, I'll do that, but just to let you know

File: libs/shared/src/lib/components/email/components/dataset-filter/dataset-filter.component.html [hunk 2]
```diff
@@ -402,34 +402,30 @@
...
-                    <ui-icon class="text-3xl" icon="delete" variant="danger">
-                    </ui-icon>
-                  </button>
+                  <ui-button
+                      [variant]="'danger'"
+                      class="hide-icons hover:text-primary-600 mt-8"
+                      [isIcon]="true"
+                      size="large"
+                      (click)="deleteDatasetFilter(fieldIndex)"
+                      [icon]="'delete'"
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/email/components/dataset-filter/dataset-filter.component.html:411]: @HasnatL-Adapptlimited 
fyi
icon="delete" is better syntax than [icon]="'delete'"

indeed, the [ ] syntax should only be preferred for dynamic inputs

I'll do the corrections while checking the PR

**Comment by AntoineRelief** [comment]
@HasnatL-Adapptlimited 

merged with some additional changes:
- more comments
- remove more useless css classes
- remove some useless classes in html ( not defined anywhere )

**Pull request merged**